kernel/timeout.c

```diff
@@ -169,29 +162,46 @@ void z_clock_announce(s32_t ticks)
 #endif
 
 	k_spinlock_key_t key = k_spin_lock(&timeout_lock);
+	struct _timeout *t = first();
 
-	announce_remaining = ticks;
+	curr_tick += ticks;
 
-	while (first() != NULL && first()->dticks <= announce_remaining) {
-		struct _timeout *t = first();
-		int dt = t->dticks;
+	if (t) {
+		sys_dlist_t ready;
+		sys_dnode_t *node;
+		s32_t announce_remaining = ticks;
```

Comment:
Maybe `ticks_passed` would be better name than `announce_remaining`. We just got the announcement...

Reply:
The variable's role is the same as `announce_remaining`, it's just no longer global.  `ticks` contains the number of ticks that has passed, so `ticks_remaining` would be a reasonable replacement to record the ticks that have not been processed.

Reply:
Sounds OK. I have bad connotations with `announce_remaining` ;-)